pkg/api/handlers.go

```diff
@@ -39,6 +43,35 @@ func newHandler(context *clusterd.Context, config *Config) *Handler {
 	}
 }
 
+func (h *Handler) Init() error {
+	var metricsConn ceph.Connection
+	var err error
+
+	// establish a connection to the ceph cluster and register our metrics exporter with prometheus
+	for {
+		metricsConn, err = h.connectToCeph()
+		if err == nil {
+			break
+		}
+
+		logger.Noticef("failed to open metrics connection to ceph cluster (will retry): %+v", err)
+		<-time.After(time.Duration(h.config.InitRetryMs) * time.Millisecond)
```

Comment:
Yes i think some sort of break from this would be an improvement.  it could probably be done in a goroutine so it doesn't prevent the rest of the API service from coming up.